cli/runner/local.go

```diff
@@ -99,10 +101,15 @@ func (r *localRunner) Setup() error {
 		return err
 	}
 
+	port, err := freeport.GetFreePort()
+	if err != nil {
+		return err
+	}
+
 	// start port forwarding
 	sockatCmd := []string{
 		"socat",
-		"tcp-listen:9222,reuseaddr,fork",
+		fmt.Sprintf("tcp-listen:%s,reuseaddr,fork", strconv.Itoa(port)),
```

Comment:
@alexplischke it is currently already as desired:

```
random host port -> port forward to 9223 on the container
```

We start Chrome with the remote debugging port set to [`9223`](https://github.com/saucelabs/sauce-puppeteer-runner/blob/594be41ee9fbbfb5e218a6622df16f172d50cad3/src/constants.js#L24). We had to use `socat` because it allows to relay the connection to outside of the container. Chrome has a security mechanism that would not allow to connect to the devtools interface from outside of the host machine. I haven't found another way around this without using socat.

Reply:
> Chrome has a security mechanism that would not allow to connect to the devtools interface from outside of the host machine. I haven't found another way around this without using socat.

Oh interesting, I had no idea! Good to know!